core/src/main/java/org/apache/calcite/rex/RexCall.java

```diff
@@ -73,6 +85,23 @@
     SIMPLE_BINARY_OPS = Sets.immutableEnumSet(kinds);
   }
 
+  private static final Set<SqlKind> SIMPLE_COMMUTATIVE_OPS;
+
+  static {
+    SIMPLE_COMMUTATIVE_OPS = Sets.immutableEnumSet(
+        SqlKind.AND, SqlKind.OR, SqlKind.PLUS, SqlKind.TIMES, SqlKind.EQUALS, SqlKind.NOT_EQUALS,
+        SqlKind.IS_DISTINCT_FROM, SqlKind.IS_NOT_DISTINCT_FROM,
+        SqlKind.GREATEST, SqlKind.LEAST);
```

Comment:
Compared to list these `SqlKind` here, i would suggest to new a fresh new `SqlKind` named `Symmetrical` in file `SqlKind.java` which include these symmetrical operators.

Reply:
Moved to `SqlKind#SYMMETRICAL` and `SqlKind#SYMMETRICAL_SAME_ARG_TYPE` (which is for cases like `PLUS`, and `TIMES` which can be re-arranged only in case the argument types are the same).
In other words, timestamp + interval vs interval + timestamp are different expressions.